soa/service/s3.cc

```diff
@@ -401,7 +401,27 @@ performSync() const
 
         myRequest.get_info(CURLINFO_RESPONSE_CODE, responseCode);
 
-        if (responseCode >= 300 && responseCode != 404) {
+        /* Detect so-called "REST error", which may or may not have an HTTP
+           status >= 200
+           (http://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html)
```

Comment:
Could this be a bug on S3 side because no where in the documentation they say that they could send a status code 200 and an error in the body.  You mentioned one case to me this morning.  If that case is documented somewhere, we should point to it.